pkg/tests/zip_test.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import filecmp
+import os
+import unittest
+from bazel_tools.tools.python.runfiles import runfiles
+from build_zip import parse_date, ZIP_EPOCH
+from zipfile import ZipFile
+
+HELLO_CRC = 2069210904
+LOREM_CRC = 2178844372
+
+class ZipTest(unittest.TestCase):
+    def get_test_zip(self, zipfile):
+        """Get the file path to a generated zip in the runfiles."""
+
+        return self.data_files.Rlocation(
+            "rules_pkg/tests/" + zipfile
```

Comment:
The archive tests used `os.join` which was admirable but didn't work out well on Windows: we really just want forward slashes here.